From 225a93422c0fa7c127433cb1dc5e82678ed84a36 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Tue, 12 Apr 2022 15:26:01 +0200
Subject: [PATCH 01/17] modify version string

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 81bb4b8e..225137f7 100755
--- a/configure
+++ b/configure
@@ -1118,7 +1118,7 @@ else
   CHRONY_VERSION="DEVELOPMENT"
 fi
 
-add_def CHRONY_VERSION "\"${CHRONY_VERSION}\""
+add_def CHRONY_VERSION "\"FUZZ ${CHRONY_VERSION}\""
 
 for f in Makefile doc/Makefile test/unit/Makefile
 do
-- 
2.41.0

