From 06e4088d8dc2c91a00507c4063125607e509c99e Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Wed, 20 Apr 2022 16:48:12 +0200
Subject: [PATCH 02/17] defer afl initialization

---
 main.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/main.c b/main.c
index 32337078..ae9d2544 100644
--- a/main.c
+++ b/main.c
@@ -460,6 +460,11 @@ int main
 
   LOG_Initialise();
 
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+  if (getenv("TEST_EARLY_AFL_INIT"))
+    __AFL_INIT();
+#endif
+
   /* Parse long command-line options */
   for (optind = 1; optind < argc; optind++) {
     if (!strcmp("--help", argv[optind])) {
@@ -663,6 +668,11 @@ int main
   TMC_Initialise();
   SMT_Initialise();
 
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+  if (!getenv("TEST_EARLY_AFL_INIT"))
+    __AFL_INIT();
+#endif
+
   /* From now on, it is safe to do finalisation on exit */
   initialised = 1;
 
-- 
2.41.0

