From 2734530392aa6854dac4ca3faec0a18908d04022 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Wed, 20 Apr 2022 16:49:21 +0200
Subject: [PATCH 03/17] catch fatal errors

---
 logging.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/logging.c b/logging.c
index 22c326c0..53aef61c 100644
--- a/logging.c
+++ b/logging.c
@@ -185,6 +185,9 @@ void LOG_Message(LOG_Severity severity,
         system_log = 0;
         log_message(1, severity, buf);
       }
+      if (getenv("TEST_FATAL_SUCCESS"))
+        exit(0);
+      assert(0);
       exit(1);
       break;
     default:
-- 
2.41.0

