From 6e821bb7b66ac905c595e7a19d547d72537d08b5 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Tue, 12 Apr 2022 16:10:01 +0200
Subject: [PATCH 05/19] disable writing and removing of files

---
 util.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/util.c b/util.c
index c4c89345..61a9e89b 100644
--- a/util.c
+++ b/util.c
@@ -1279,11 +1279,13 @@ UTI_OpenFile(const char *basedir, const char *name, const char *suffix,
       break;
     case 'w':
     case 'W':
+      return NULL;
       flags = O_WRONLY | O_CREAT | O_EXCL;
       file_mode = "w";
       break;
     case 'a':
     case 'A':
+      return NULL;
       flags = O_WRONLY | O_CREAT | O_APPEND | O_NOFOLLOW;
       file_mode = "a";
       break;
@@ -1368,6 +1370,9 @@ UTI_RemoveFile(const char *basedir, const char *name, const char *suffix)
     return 0;
   }
 
+  DEBUG_LOG("Refused to remove %s", path);
+  return 0;
+
   if (unlink(path) < 0) {
     LOG(LOGS_ERR, "Could not remove %s : %s", path, strerror(errno));
     return 0;
-- 
2.35.1

