From d26d63c99cc1be169dd2cff17acf33ccb20d4fde Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Tue, 12 Apr 2022 16:16:39 +0200
Subject: [PATCH 06/17] allow files to be switched to stdin

---
 util.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/util.c b/util.c
index d3a353d6..cb0d8750 100644
--- a/util.c
+++ b/util.c
@@ -1332,7 +1332,7 @@ UTI_OpenFile(const char *basedir, const char *name, const char *suffix,
              char mode, mode_t perm)
 {
   const char *file_mode;
-  char path[PATH_MAX];
+  char path[PATH_MAX], *env;
   LOG_Severity severity;
   int fd, flags;
   FILE *file;
@@ -1342,6 +1342,12 @@ UTI_OpenFile(const char *basedir, const char *name, const char *suffix,
   if (!join_path(basedir, name, suffix, path, sizeof (path), severity))
     return NULL;
 
+  env = getenv("TEST_STDIN_FILE");
+  if (env && strcmp(env, path) == 0) {
+    DEBUG_LOG("Opened %s as stdin", path);
+    return stdin;
+  }
+
   switch (mode) {
     case 'r':
     case 'R':
-- 
2.41.0

