From fcc4feccf196210639bce4223da11eb26c83692e Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Wed, 20 Apr 2022 16:50:36 +0200
Subject: [PATCH 08/19] always return success with -t option

---
 main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/main.c b/main.c
index 1500654f..538ee6b4 100644
--- a/main.c
+++ b/main.c
@@ -171,6 +171,7 @@ quit_timeout(void *arg)
 
   /* Return with non-zero status if the clock is not synchronised */
   exit_status = REF_GetOurStratum() >= NTP_MAX_STRATUM;
+  exit_status = 0;
   SCH_QuitProgram();
 }
 
-- 
2.37.3

