From 2f4679276c7ddaeef875d2f808a61eb225d29b1d Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Wed, 20 Apr 2022 16:51:30 +0200
Subject: [PATCH 10/19] reduce maximum chronyc retries and timeout

---
 client.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/client.c b/client.c
index 9201b76d..44890646 100644
--- a/client.c
+++ b/client.c
@@ -2908,6 +2908,9 @@ process_cmd_waitsync(char *line)
   if (sscanf(line, "%d %lf %lf %lf", &max_tries, &max_correction, &max_skew_ppm, &interval))
     ;
 
+  max_tries = CLAMP(1, max_tries, 3);
+  interval = CLAMP(0.0, max_tries, 10.0);
+
   /* Don't allow shorter interval than 0.1 seconds */
   if (interval < 0.1)
     interval = 0.1;
@@ -2979,7 +2982,7 @@ process_cmd_timeout(const char *line)
     LOG(LOGS_ERR, "Timeout %d is too short", timeout);
     return 0;
   }
-  initial_timeout = timeout;
+  initial_timeout = MIN(timeout, 10000);
   return 1;
 }
 
@@ -2991,7 +2994,7 @@ process_cmd_retries(const char *line)
   int retries;
 
   retries = atoi(line);
-  if (retries < 0 || retries > 30) {
+  if (retries < 0 || retries > 5) {
     LOG(LOGS_ERR, "Invalid maximum number of retries");
     return 0;
   }
-- 
2.36.1

