From e172b910109c292a60c21c1cc62748555f461261 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Tue, 19 Apr 2022 13:34:51 +0200
Subject: [PATCH 11/19] don't stop in chronyc on error

---
 client.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/client.c b/client.c
index 44890646..d29dc970 100644
--- a/client.c
+++ b/client.c
@@ -3278,6 +3278,7 @@ process_args(int argc, char **argv, int multi)
       continue;
 
     ret = process_line(line);
+    ret = 1;
     if (!ret || quit)
       break;
 
@@ -3435,6 +3436,7 @@ main(int argc, char **argv)
       line = read_line();
       if (line && !quit) {
         ret = process_line(line);
+        ret = 1;
       }else {
 	/* supply the final '\n' when user exits via ^D */
         if( on_terminal ) printf("\n");
-- 
2.36.1

