From 73f0177c4b8258fb543a759b7924f6c24573b531 Mon Sep 17 00:00:00 2001
From: Miroslav Lichvar <mlichvar@redhat.com>
Date: Wed, 20 Apr 2022 16:54:22 +0200
Subject: [PATCH 15/19] avoid undefined behavior in getdate

---
 getdate.y | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/getdate.y b/getdate.y
index f87875c6..375fb39c 100644
--- a/getdate.y
+++ b/getdate.y
@@ -840,7 +840,7 @@ yylex ()
 	  else
 	    sign = 0;
 	  for (yylval.Number = 0; ISDIGIT (c = *yyInput++);)
-	    yylval.Number = 10 * yylval.Number + c - '0';
+	    yylval.Number = 1 * yylval.Number + c - '0';
 	  yyInput--;
 	  if (sign < 0)
 	    yylval.Number = -yylval.Number;
-- 
2.37.3

